engine/runtime-parser/src/main/scala/org/enso/compiler/core/ir/IrLazyImMap.scala

```diff
@@ -1,55 +1,43 @@
 package org.enso.compiler.core.ir
 
-import scala.collection.immutable.AbstractMap
+import scala.collection.immutable.StrictOptimizedMapOps
 import scala.collection.immutable.Map
-import scala.collection.immutable.MapOps
-import scala.collection.IterableFactory
-import scala.collection.immutable.Iterable
-import scala.collection.MapFactory
-import scala.annotation.unchecked.uncheckedVariance
-import scala.collection.mutable.Builder
+import scala.collection.Iterator
+
+final private class IrLazyImMap[K, V](
+  val underlying: java.util.Map[K, V]
```

Comment:
Wow, thanks @hubertp for this code change! Looks like it is possible to have an _immutable Scala map_ that delegates to `java.util.Map`. Kudos to @thurka for coming up with this delegation idea. Thank you guys!